model/cluster_utility.go

```diff
@@ -169,6 +172,76 @@ func (h *UtilityGroupVersions) AsMap() map[string]*HelmUtilityVersion {
 	}
 }
 
+// UnmarshalJSON is a custom JSON unmarshaler that can handle both the
+// old Version string type and the new type. It is entirely
+// self-contained, including types, so that it can be easily removed
+// when no more clusters exist with the old version format.
+// TODO DELETE THIS
+func (h *UtilityGroupVersions) UnmarshalJSON(bytes []byte) error {
+	type innerUtilityGroupVersions struct {
+		PrometheusOperator  *HelmUtilityVersion
+		Thanos              *HelmUtilityVersion
+		Nginx               *HelmUtilityVersion
+		NginxInternal       *HelmUtilityVersion
+		Fluentbit           *HelmUtilityVersion
+		Teleport            *HelmUtilityVersion
+		Pgbouncer           *HelmUtilityVersion
+		Promtail            *HelmUtilityVersion
+		Rtcd                *HelmUtilityVersion
+		NodeProblemDetector *HelmUtilityVersion
+		MetricsServer       *HelmUtilityVersion
+		Velero              *HelmUtilityVersion
+		Cloudprober         *HelmUtilityVersion
+	}
+
+	var versions *innerUtilityGroupVersions = &innerUtilityGroupVersions{}
+
+	err := json.Unmarshal(bytes, versions)
+	if err != nil && strings.Contains(err.Error(), "cannot unmarshal string into Go struct field") {
```

Comment:
Just a thought. If we do not need to know the version of the utilities of the deleted cluster, we can simply ignore the error and return `nil`.

Reply:
I'm ok with this, but previous to the discussion I'd rather submit the fix in case we want the data for some strange reason.